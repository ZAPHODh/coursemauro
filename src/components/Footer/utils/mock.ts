export const medias = [
    {
        media: 'Instagram',
        page: 'https://www.instagram.com/maurochrisostimo_?igsh=bHR0enlmdGlveG9w',
    },
    {
        media: 'Tiktok',
        page: 'https://www.tiktok.com/@maurochrisostimo?_t=8nMAal30P9a&_r=1',
    },
    {
        media: 'Whatsapp',
        page: 'https://api.whatsapp.com/send?phone=5521967147005&text=Ol√°,%20gostaria%20de%20saber%20mais%20sobre%20o%20curso',
    },
]
