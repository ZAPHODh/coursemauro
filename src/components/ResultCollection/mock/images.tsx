import mauroPic from '../../../../public/images/mauro.webp'

export const images = [
    { img: mauroPic, alt: 'test' },
    { img: mauroPic, alt: 'test' },
    { img: mauroPic, alt: 'test' },
    { img: mauroPic, alt: 'test' },
    { img: mauroPic, alt: 'test' },
]
