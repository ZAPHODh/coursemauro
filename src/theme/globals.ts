'use client'

import { Montserrat } from 'next/font/google'
import { createGlobalStyle } from 'styled-components'
import { css } from 'styled-components'

const montserrat = Montserrat({
    subsets: ['latin'],
    fallback: ['Georgia', 'ui-serif', 'serif'],
    weight: ['300', '700'],
})

export const GlobalStyles = createGlobalStyle`${() => css`
    html {
        font-family: ${montserrat.style.fontFamily};
        max-width: 100vw;
    }
    ,
    main {
        max-width: 100vw;
    }
    body {
        padding: 0;
        margin: 0;
        overflow: hidden;
    }
    ,
    input {
        font-family: inherit;
    }
    ,
    a {
        color: inherit;
    }
    ,
    * {
        box-sizing: border-box;
    }
    ,
    button {
        font-family: inherit;
    }
`}`
