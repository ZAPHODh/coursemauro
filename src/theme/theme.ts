import { DefaultTheme } from 'styled-components'

export const theme: DefaultTheme = {
    colors: {
        white: '#ffffff ',
        darker: '#0B0B0D',
        dark: '#181619',
        normalDark: '#181B22',
        brightDark: '#272A31',
        gold: '#A76B09',
        brightGold: '#DDA74F',
    },
}
