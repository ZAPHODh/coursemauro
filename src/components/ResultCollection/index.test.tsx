import { render } from '@testing-library/react'
import { ResultCollection } from '.'
import { ThemeClient } from '../../theme/ThemeClient'
import { images } from './mock/images'

describe('ResultCollection', () => {
    it('should render the ResultCollection', () => {
        const { container, debug } = render(
            <ThemeClient>
                <ResultCollection results={images} title="test" />
            </ThemeClient>
        )
        expect(container).toBeInTheDocument()
    })
    it('should render the correct number of images', () => {
        const { container } = render(
            <ThemeClient>
                <ResultCollection results={images} title="test" />
            </ThemeClient>
        )
        const imagesElements = container.querySelectorAll('img')

        expect(imagesElements.length).toBe(images.length)
    })
    it('should take a ResultCollection snapshot', () => {
        const { container } = render(
            <ThemeClient>
                <ResultCollection results={images} title="test" />
            </ThemeClient>
        )

        expect(container).toMatchSnapshot()
    })
})
