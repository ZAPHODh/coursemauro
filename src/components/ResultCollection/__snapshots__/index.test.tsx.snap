// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCollection should take a ResultCollection snapshot 1`] = `
<div>
  <section
    aria-label="Coleção de Imagens"
    class="sc-94e3b9c1-0 gXixWR"
  >
    <h2
      class="sc-e26ab1cd-0 cioyZq"
      padding="10px"
    >
      test
    </h2>
    <div
      class="sc-94e3b9c1-1 gBMHrs"
    >
      <img
        alt="test"
        class="sc-73d527aa-0 fHlBsz"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/img.jpg"
        srcset="/img.jpg 1x, /img.jpg 2x"
        style="color: transparent;"
        width="40"
      />
      <img
        alt="test"
        class="sc-73d527aa-0 fHlBsz"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/img.jpg"
        srcset="/img.jpg 1x, /img.jpg 2x"
        style="color: transparent;"
        width="40"
      />
      <img
        alt="test"
        class="sc-73d527aa-0 fHlBsz"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/img.jpg"
        srcset="/img.jpg 1x, /img.jpg 2x"
        style="color: transparent;"
        width="40"
      />
      <img
        alt="test"
        class="sc-73d527aa-0 fHlBsz"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/img.jpg"
        srcset="/img.jpg 1x, /img.jpg 2x"
        style="color: transparent;"
        width="40"
      />
      <img
        alt="test"
        class="sc-73d527aa-0 fHlBsz"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/img.jpg"
        srcset="/img.jpg 1x, /img.jpg 2x"
        style="color: transparent;"
        width="40"
      />
    </div>
  </section>
</div>
`;
